adam/learner/objects.py

```diff
@@ -443,39 +483,48 @@ def learn_from(
         pass
 
     def enrich_during_learning(
-        self,
-        language_perception_semantic_alignment: LanguagePerceptionSemanticAlignment,
-        language_generator: LanguageGenerator[
-            HighLevelSemanticsSituation, LinearizedDependencyTree
-        ] = GAILA_PHASE_1_LANGUAGE_GENERATOR,
+        self, language_perception_semantic_alignment: LanguagePerceptionSemanticAlignment
     ) -> LanguagePerceptionSemanticAlignment:
         return self._object_recognizer.match_objects_with_language(
-            language_perception_semantic_alignment, language_generator=language_generator
+            language_perception_semantic_alignment
         )
 
     def enrich_during_description(
         self, perception_semantic_alignment: PerceptionSemanticAlignment
     ) -> PerceptionSemanticAlignment:
         (new_perception_semantic_alignment, _) = self._object_recognizer.match_objects(
-            perception_semantic_alignment, language_generator=self._language_generator
+            perception_semantic_alignment
         )
         return new_perception_semantic_alignment
 
     def templates_for_concept(self, concept: Concept) -> ImmutableSet[SurfaceTemplate]:
-        if self._language_generator == GAILA_PHASE_1_LANGUAGE_GENERATOR:
+        if self._language_mode == LanguageMode.ENGLISH:
             return self._concepts_to_templates[concept]
-        else:
+        elif self._language_mode == LanguageMode.CHINESE:
             mappings = (
                 GAILA_PHASE_1_CHINESE_LEXICON._ontology_node_to_word  # pylint:disable=protected-access
             )
             for k, v in mappings.items():
                 if k.handle == concept.debug_string:
-                    return immutableset([SurfaceTemplate.for_object_name(v.base_form)])
+                    return immutableset(
+                        [
+                            SurfaceTemplate.for_object_name(
+                                v.base_form, language_mode=self._language_mode
+                            )
+                        ]
+                    )
         raise RuntimeError(f"Invalid concept {concept}")
 
     def log_hypotheses(self, log_output_path: Path) -> None:
         pass
 
+    def _intersect_hypothesis(
+        self,
+        previous_pattern_hypothesis: PerceptionGraphTemplate,
+        current_pattern_hypothesis: PerceptionGraphTemplate,
+    ) -> Optional[PerceptionGraphTemplate]:
+        pass
```

Comment:
Is this what you mean to have here?

Reply:
Yes, the `ObjectRecognizerAsObjectLearner` does not learn anything about its patterns so this function never actually gets called but it is defined as an abstract method in a base class. I'll add an explanatory comment here.

Reply:
@LichMaster98 : Thanks. Perhaps also change the content to an exception, since if it ever does get called we know something has gone wrong?